test/unit/io/alignment_file/format_bam_test.cpp

```diff
@@ -516,3 +519,38 @@ TEST_F(bam_format, too_long_cigar_string_write)
 
     EXPECT_TRUE(os.str() == expected); // do not use EXPECT_EQ because if this fails the output will be huge :D
 }
+
+// https://github.com/seqan/seqan3/issues/2417
+TEST_F(bam_format, issue2417)
+{
+    std::string const input{
+        // @HD	VN:1.6
+        // @SQ	SN:ref	LN:1904
+        // read1	117	ref	1	0	*	=	1	0	ACGTA	IIIII
+        '\x42', '\x41', '\x4D', '\x01', '\x1E', '\x00', '\x00', '\x00', '\x40', '\x48', '\x44', '\x09', '\x56', '\x4E',
+        '\x3A', '\x31', '\x2E', '\x36', '\x0A', '\x40', '\x53', '\x51', '\x09', '\x53', '\x4E', '\x3A', '\x72', '\x65',
+        '\x66', '\x09', '\x4C', '\x4E', '\x3A', '\x31', '\x39', '\x30', '\x34', '\x0A', '\x01', '\x00', '\x00', '\x00',
+        '\x04', '\x00', '\x00', '\x00', '\x72', '\x65', '\x66', '\x00', '\x70', '\x07', '\x00', '\x00', '\x2E', '\x00',
+        '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x06', '\x00', '\x49', '\x12',
+        '\x00', '\x00', '\x75', '\x00', '\x05', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00',
+        '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x72', '\x65', '\x61', '\x64', '\x31', '\x00', '\x12', '\x48',
+        '\x10', '\x28', '\x28', '\x28', '\x28', '\x28'
+    };
+
+    std::istringstream stream{input};
+
+    seqan3::alignment_file_input fin{stream, seqan3::format_bam{}, seqan3::fields<seqan3::field::id,
+                                                                                  seqan3::field::alignment>{}};
+
+    std::vector<seqan3::gapped<seqan3::dna5>> const empty_sequence{};
+
+    // In 2417, the sequence was not consumed. Thus, wrong bytes were read for the following records.
+    // With the chosen `input` this also means that there will be more than 1 record in the alignment file.
+    // Hence, we need the for loop even though there is only 1 record.
```

Comment:
I don't understand this remark.

Reply:
There's only one record in the bam file.
The bug caused the sequence to not be consumed, which also means that we think there is another record because we are not at the end of the stream after reading one record.
The second record will then be gibberish and throw something somewhere while being parsed.

Since I only read `id` and `alignment`, there is not really something I can check that would be parsed incorrectly in this one record (`id` is before the sequence and alignment is an empty dummy).

I could also check if the file iterator is at end after reading one record, but the for-loop is closer to the reported bug/use-case and also kinda easier to parse IMO.

Reply:
Okay! Can you add a counter that counts the number of loop-passes? That one should be 1 right?